functional-tests/run_tests.sh

```diff
@@ -1,8 +1,12 @@
 #! /bin/bash
 set -exv -o pipefail
 
-# NOTE: FIREFOX HEADLESS IS THE ONLY BROWSER WHERE THE UPLOAD DOWNLOAD TEST CAN BE COMPLETED
+# Dry run project to cache gradle and dependencies
+./gradlew -m
```

Comment:
Separated the step to actually cache gradle deps from running tests to isolate failures better